File: arrays/go/array3.go
package main

import "fmt"

func main(){
	arr1:=[]int{1,1,2,3,4,5}
	arr2:=[]int{2,3,4,4,5,6}
	unionOfArray(arr1, arr2)
}

//Union using two Pointer method.
func unionOfArray(arr1 []int, arr2 []int){
	n1:=len(arr1)
	n2:=len(arr2)
	j:=0
	i:=0
	union:=make([]int,0)
	
	for i<n1 && j<n2 {
		if arr1[i]<=arr2[j] { 
			if len(union)==0 || union[len(union)-1]!=arr1[i] {
				union=append(union, arr1[i]) 
			}
			i++
		}else{
			if len(union)==0 || union[len(union)-1]!=arr2[j]{
				union=append(union, arr2[j]) 
			}
			j++
		}
	}

	for i<n1{
			if len(union)==0 || union[len(union)-1]!=arr1[i] {
				union=append(union, arr1[i]) 
			}
			i++
		}
		
	for j<n2{
			if len(union)==0 || union[len(union)-1]!=arr2[j]{
				union=append(union, arr2[j]) 
			}
			j++
	}

	fmt.Println(union)
}

File: arrays/python/array3.py
arr1=[1,1,2,3,4,5]
arr2=[2,3,4,4,5,6]

def union_of_array(arr1, arr2):
    n1 = len(arr1)
    n2 = len(arr2)
    union = []
    i, j = 0, 0
    while i<n1 and j<n2:
        if arr1[i]<=arr2[j]:
            if not union or union[-1]!=arr1[i]: union.append(arr1[i])
            i+=1
        else: 
            if not union or union[-1]!=arr2[j]: union.append(arr2[j])
            j+=1
    while i<n1:
        if not union or union[-1]!=arr1[i]: union.append(arr1[i])
        i+=1
    while j<n2:
        if not union or union[-1]!=arr2[j]: union.append(arr2[j])
        j+=1
    print(union)

union_of_array(arr1, arr2)